Sources/ParseSwift/Objects/ParseUser.swift

```diff
@@ -41,27 +41,22 @@ struct CurrentUserContainer<T: ParseUser>: Codable {
 // MARK: Current User Support
 extension ParseUser {
     static var currentUserContainer: CurrentUserContainer<Self>? {
-        get {
-            guard let currentUserInMemory: CurrentUserContainer<Self>
-                = try? ParseStorage.shared.get(valueFor: ParseStorage.Keys.currentUser) else {
-                return try? KeychainStore.shared.get(valueFor: ParseStorage.Keys.currentUser)
-            }
-            return currentUserInMemory
-        }
-        set { try? ParseStorage.shared.set(newValue, for: ParseStorage.Keys.currentUser) }
+        get { try? ParseStorage.shared.secureStore.get(valueFor: ParseStorage.Keys.currentUser) }
+        set { try? ParseStorage.shared.secureStore.set(newValue, for: ParseStorage.Keys.currentUser)}
     }
 
-    internal static func saveCurrentContainerToKeychain() {
-        //Only save the BaseParseUser to keep Keychain footprint finite
-        guard let currentUserInMemory: CurrentUserContainer<BaseParseUser>
-            = try? ParseStorage.shared.get(valueFor: ParseStorage.Keys.currentUser) else {
+    internal static func saveCurrentUserContainer() {
+        //Only save the BaseParseUser to keep memory footprint finite
+        guard let currentUser: CurrentUserContainer<BaseParseUser> =
```

Comment:
This is similar to my previous comment, basically all changes are permanent before they are actually saved to the parse-server with no way to revert